src/components/common/CodeEditor.js

```diff
@@ -0,0 +1,293 @@
+  DEFAULT_WARNING_COLOR,
+  PUBLIC_VISIBILITY,
+} from '../../config/settings';
+
+const styles = (theme) => ({
+  container: {
+    borderSpacing: 0,
+    maxWidth: '800px',
+    width: '90%',
+    margin: '20px auto auto',
+    paddingLeft: '20px',
+    boxSizing: 'border-box',
+    paddingRight: '0px',
+  },
+  editor: {
+    minHeight: '100px',
+    height: '100%',
+    width: '100%',
+  },
+  button: {
+    marginRight: theme.spacing(),
+  },
+  noFlex: {
+    display: 'block',
+  },
+  paper: {
+    padding: theme.spacing(1),
+  },
+  divider: {
+    marginBottom: theme.spacing(2),
+  },
+  commitTextField: {
+    '& .MuiFormLabel-root.Mui-error, .MuiFormHelperText-root.Mui-error, ': {
+      color: DEFAULT_WARNING_COLOR,
+    },
+    '& .MuiOutlinedInput-root.Mui-error .MuiOutlinedInput-notchedOutline': {
+      borderColor: DEFAULT_WARNING_COLOR,
+    },
+  },
+});
+
+class CodeEditor extends React.Component {
+  static propTypes = {
+    classes: PropTypes.shape({
+      container: PropTypes.string,
+      editor: PropTypes.string,
+      button: PropTypes.string,
+      noFlex: PropTypes.string,
+      paper: PropTypes.string,
+      divider: PropTypes.string,
+      commitTextField: PropTypes.string,
+    }).isRequired,
+    t: PropTypes.func.isRequired,
+    dispatchCloseEditorView: PropTypes.func.isRequired,
+    dispatchPostAppInstanceResource: PropTypes.func.isRequired,
+    // dispatchSetCodeEditorSettings: PropTypes.func.isRequired,
+    commit: PropTypes.shape({
+      code: PropTypes.string.isRequired,
+      commitMessage: PropTypes.string.isRequired,
+      commitDescription: PropTypes.string.isRequired,
+    }).isRequired,
+    programmingLanguage: PropTypes.string.isRequired,
+    userId: PropTypes.string.isRequired,
+    maxMessageLength: PropTypes.number,
+  };
+
+  static defaultProps = {
+    maxMessageLength: DEFAULT_MAX_COMMIT_MESSAGE_LENGTH,
+  };
+
+  state = (() => {
+    const { commit } = this.props;
+    return {
+      commit,
+    };
+  })();
+
+  handleChangeEditor = (key) => (value) => {
+    this.setState((prevState) => {
+      // get the previous state's settings
+      const commit = { ...prevState.commit };
+      // use lodash to be able to use dot and array notation
+      _.set(commit, key, value);
+      return { commit };
+    });
+  };
+
+  handleChangeTextField =
+    (key) =>
+    ({ target: { value } }) => {
+      this.setState((prevState) => {
+        // get the previous state's settings
+        const commit = { ...prevState.commit };
+        // use lodash to be able to use dot and array notation
+        _.set(commit, key, value);
+        return { commit };
+      });
+    };
+
+  handleCommit = () => {
+    const { dispatchPostAppInstanceResource, userId } = this.props;
+    const { commit } = this.state;
+    // post app instance resource as public
+    dispatchPostAppInstanceResource({
+      data: commit,
+      type: CODE,
+      visibility: PUBLIC_VISIBILITY,
```

Comment:
This should have the public/private switch.

Reply:
Yes i am adding it in the next commit